### ISSUE TITLE
Use logger instead of print

### ISSUE THREAD

**Issue by tkarna** [thetisproject/thetis#39]
Use python logging module to handle all print statements etc.

**Comment by tkarna** [comment]
Any ideas how we should implement logging in thetis now that firedrake logger has landed?

I guess one option is to just use firedrake's `info` and `warning` etc functions. Alternative is to add a new thetis logger, similar to firedrake's (we'd need similar parallel handling for instance).

Personally I'd prefer that normal info messages would not have a header (e.g. `firedrake.info` prints `firedrake:INFO ...`).

**Comment by wence-** [comment]
I'm in favour of using a separate logger (so you can distinguish messages, and hunt down where they're from).  If you want to use the same parallel handling code that we use in firedrake, we can change it to allow registering arbitrary packages.  But it's not very much code.  I'd rather firedrake not be a one-stop-shop for providing logging facilities if you want separation of package logging.  So maybe something like:

```python
import logging
from thetis.utility import COMM_WORLD

logger = logging.getLogger("thetis")

for handler in logger.handlers:
    logger.removeHandler(handler)

handler = logging.StreamHandler()
handler.setFormatter(logging.Formatter("%(message)s"))

if COMM_WORLD.rank != 0:
    handler = logging.NullHandler()

logger.addHandler(handler)
```

**Comment by tkarna** [comment]
Sounds like a good approach. I'll start implementing this.

**Comment by tkarna** [comment]
A candidate is now implemented in #58.
